drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -321,56 +309,42 @@ local function temp_attr_handler(driver, device, ib, response)
-      return
-    end
-    local temp = utils.round(CONVERSION_CONSTANT/ib.data.value)
-    device:set_field(COLOR_TEMP_MIN, temp)
-    local color_temp_max = device:get_field(COLOR_TEMP_MAX)
-    if color_temp_max then
-      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorTemperature.colorTemperatureRange({minimum = temp, maximum = color_temp_max}))
-      device:set_field(COLOR_TEMP_MIN, nil)
-      device:set_field(COLOR_TEMP_MAX, nil)
+local mired_bounds_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value ~= nil then
+      if (ib.data.value < COLOR_TEMPERATURE_MIRED_MIN or ib.data.value > COLOR_TEMPERATURE_MIRED_MAX) then
+        device.log.warn_with({hub_logs = true}, string.format("Device reported a min or max color temperature %d mired outside of supported capability range", ib.data.value))
+        return
+      end
+      local temp_in_kelvin = mired_to_kelvin(ib.data.value)
+      device:set_field(COLOR_TEMP_BOUND_RECEIVED..minOrMax, temp_in_kelvin)
+      local min = device:get_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN)
+      local max = device:get_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX)
+      if min ~= nil and max ~= nil and min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorTemperature.colorTemperatureRange({ value = {minimum = min, maximum = max} }))
+        device:set_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX, nil)
+        device:set_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN, nil)
+      end
     end
   end
 end
 
-local function color_temp_max_attr_handler(driver, device, ib, response)
-  if ib.data.value ~= nil then
-    if (ib.data.value < COLOR_TEMPERATURE_MIRED_MIN or ib.data.value > COLOR_TEMPERATURE_MIRED_MAX) then
-      device.log.warn_with({hub_logs = true}, string.format("Device reported maximum color temperature %d mired outside of supported capability range", ib.data.value))
-      return
-    end
-    local temp = utils.round(CONVERSION_CONSTANT/ib.data.value)
-    device:set_field(COLOR_TEMP_MAX, temp)
-    local color_temp_min = device:get_field(COLOR_TEMP_MIN)
-    if color_temp_min then
-      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorTemperature.colorTemperatureRange({minimum = color_temp_min, maximum = temp}))
-      device:set_field(COLOR_TEMP_MIN, nil)
-      device:set_field(COLOR_TEMP_MAX, nil)
+local level_bounds_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value ~= nil then
+      local level = math.floor((ib.data.value / 254.0 * 100) + 0.5)
+      device:set_field(LEVEL_BOUND_RECEIVED..minOrMax, level)
+      local min = device:get_field(LEVEL_BOUND_RECEIVED..LEVEL_MIN)
+      local max = device:get_field(LEVEL_BOUND_RECEIVED..LEVEL_MAX)
+      if min ~= nil and max ~= nil and min < max then
```

Comment:
If we get something outside of our expected bounds we should probably log it. Also, is `min == max` a valid scenario?

Reply:
`min == max` shouldn't be valid, no

Reply:
I don't think it's very likely, but my read of the spec is that they are allowed to be equal

Reply:
I consider "allowed by the spec" and "allowed by us" to be two different things, frankly. I think we'd do better to stick to a default presentation than to show a slider with only one value.

Reply:
I addressed this in https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1220/commits/f04cf2ec2c69373a72d337e4376c0565baa95f56